addons/website_slides/models/slide_slide.py

```diff
@@ -1286,6 +1286,28 @@ def _default_website_meta(self):
     # Data / Misc
     # ---------------------------------------------------------
 
+    def _is_category_completed(self):
+        if not self.category_id:
+            return False
+        return all(self.category_id.slide_ids.mapped('user_has_completed'))
+
+    def _get_next_category_id(self):
```

Comment:
By the way, maybe "_get_next_category" and return a record instead of an ID ? Always better to deal with records :)